controllers/mirrorpeer_controller.go

```diff
@@ -715,6 +728,32 @@ func (r *MirrorPeerReconciler) processManagedClusterAddon(ctx context.Context, m
 	logger := r.Logger.With("MirrorPeer", mirrorPeer.Name)
 	logger.Info("Processing ManagedClusterAddons for MirrorPeer")
 
+	clusterManagementAddOn := addonapiv1alpha1.ClusterManagementAddOn{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: setup.TokenExchangeName,
+		},
+	}
+
+	logger.Info("Creating ClusterManagementAddOn for ManagedClusterAddons", "Name", clusterManagementAddOn.Name)
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, &clusterManagementAddOn, func() error {
+		annotations := make(map[string]string)
```

Comment:
```suggestion
		 if clusterManagementAddOn.Annotations == nil {
        clusterManagementAddOn.Annotations = make(map[string]string)
    }
```
Avoid overwriting annotations